README.md

```diff
@@ -1,46 +1,429 @@
+
+#### Schemes
+* SchemeRegistrar
+* UpgradeScheme
+* GlobalConstraintRegistrar
+* ContributionReward
+* VoteInOrganizationScheme
+* VestingScheme
+
+#### Voting Machines
+* AbsoluteVote
+* GenesisProtocol
+
+#### Global Constraints
+
+* TokenCapGC
+
+#### Others
+* DaoCreator
+
+You can pull those categorizations into your code as follows:
+
+```javascript
+import * as ArcJs from '@daostack/arc.js';
+const arcJsWrappers = await ArcJs.getDeployedContracts();
+```
+
+Note `getDeployedContracts()` does not currently cache its results and is fairly time-consuming to run, so best to cache the results yourself.
+
+See also [Obtain a wrapper from getDeployedContracts](#obtain-a-wrapper-from-getdeployedcontracts) and the following sections for several ways of instantiating Arc.js contract wrappers.
+
+### Obtain a DAO's Scheme
+
+You may obtain the wrapper for a scheme that is registered with a DAO using `DAO.getScheme`.  Here is how to get the wrapper for the `UpgradeScheme` registered with a DAO:
+
+```javascript
+const upgradeScheme = await myDao.getScheme("UpgradeScheme");
+```
+
+### Obtain a wrapper using the wrapper's factory class 
+
+Each wrapper has a factory that provides static `.new()`, `.deployed()` and `.at()` methods.  These methods are implemented by `ExtendTruffleContract`.
+
+Examples of their use:
+
+#### deployed
+
+Obtain the instance of the contract as deployed by Arc.js:
+
+```javascript
+import { UpgradeScheme } from "@daostack/arc.js";
+const deployedContract = await UpgradeScheme.deployed();
+```
+
+#### new
+
+Call `new()` to migrate a new instance of the contract:
+
+```javascript
+import { UpgradeScheme } from "@daostack/arc.js";
+const newInstance = await UpgradeScheme.new();
+```
+
+#### at
+
+Obtain the wrapper from a given address:
+
+```javascript
+import { UpgradeScheme } from "@daostack/arc.js";
+const newInstance = await UpgradeScheme.at(anAddress);
+```
+
+### Obtain a wrapper from getDeployedContracts
+
+Recall from [Categories of Arc Contracts](#categories-of-arc-contracts) that `getDeployedContracts()` returns categories of Arc contract wrappers.   You can use this information to obtain the wrappers themselves:
+
+```javascript
+import * as ArcJs from '@daostack/arc.js';
+const arcJsWrappers = ArcJs.getDeployedContracts();
+const contractAddress = arcJsWrappers.allContracts.UpgradeScheme.address;
+const contractFactory = arcJsWrappers.allContracts.UpgradeScheme.contract;
+const upgradeScheme = await contractFactory.contract.at(contractAddress);
+```
+
+### Obtain any Arc contract using Utils.requireContract
+
+Not all Arc contracts have been given wrapper classes, for example, `Avatar`, `UController` and many more.  But you can still obtain a raw TruffleContract enabling you to work with these contracts:
+
+```javascript
+import { Utils } from "@daostack/arc.js";
+const truffleContract = await Utils.requireContract("Avatar");
+```
+
+## Working with Arc.js Scripts
```

Comment:
seems like a duplicate of above

Reply:
I'm sorry, I'm not finding the duplication

Reply:
the section on ### Setting up a Testnet with Arc Contracts. seems like we should just cover all the package scripts right there, up front

Reply:
I didn't actually describe any new scripts here, rather just showed how to reference any of the scripts from the context of an application.  I don't ever plan to document *all* of the scripts in this readme.  Some folks are already complaining that the readme is too long.  Right now the remainder of the scripts are documented in package-script.js, as described in the readme.  At some point they may end up in a separate .MD file in a docs folder.

As for putting this information closer to the top:  As I also described above, my goal here is to keep things simple at the top, like a fundamental workflow that is easy to follow and read through progressively.  I want to keep special cases and more detailed information closer to the bottom.  This is not meant to be like a book with a chapter on each subject.  It is meant to be an easy-to-read, easy-to-follow getting-started introduction to the library, with some more specific reference sections later on thrown in.  I don't want to bog reader down in unnecessary details when they are just getting started.

Does that work for you?

Reply:
sure thing :)